### PR TITLE
Remove Step and Calculation

### PR DESCRIPTION
**Pull request by minotogna**
## Description

- #### Removed Steps and Calculations
   
- #### Added Chain redux store

- #### Chain is persisted without save button

## Types of changes

- [ ] Docs change
- [X] Refactoring
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How has this been tested

Tests will come later on...

#### Do you consider this PR needs further testing?

- [ ] No
- [X] Yes

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-g56hay Remove step/calculation](https://app.clickup.com/t/g56hay)

File: server/modules/analysis/service/update.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import * as ActivityLogRepository from '@server/modules/activityLog/repository/activityLogRepository'
+import * as ChainRepository from '@server/modules/analysis/repository/chain'
+import { markSurveyDraft } from '@server/modules/survey/repository/surveySchemaRepositoryUtils'
+
+export const update = async ({ user, surveyId, chain }) => {
+  const chainUuid = Chain.getUuid(chain)
+
+  return DB.tx(async (t) => {
+    const chainDb = await ChainRepository.fetchChain({ surveyId, chainUuid: Chain.getUuid(chain) }, t)
```
Comment (SteRiccio) [inline server/modules/analysis/service/update.js:15]: you could use the constant chainUuid declared above

File: server/modules/analysis/service/update.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      const type = ActivityLog.type.chainPropUpdate
+      return ActivityLogRepository.insert(user, surveyId, type, content, false, t)
+    })
+
+    // chain props and validation update
+    const fields = {
+      [TableChain.columnSet.props]: propsToUpdate,
+      [TableChain.columnSet.validation]: Chain.getValidation(chain),
+    }
+    const params = { surveyId, chainUuid, dateModified: true, fields }
```
Comment (SteRiccio) [inline server/modules/analysis/service/update.js:30]: what is _dateModified_?

File: webapp/store/ui/chain/actions/deleteChain.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { SurveyActions, SurveyState } from '@webapp/store/survey'
+
+export const deleteChain = ({ chain, history }) => async (dispatch, getState) => {
+  const state = getState()
+  const surveyId = SurveyState.getSurveyId(state)
+
+  const action = async () => {
+    dispatch(LoaderActions.showLoader())
+
+    await axios.delete(`/api/survey/${surveyId}/chain/${chain.uuid}`)
```
Comment (SteRiccio) [inline webapp/store/ui/chain/actions/deleteChain.js:14]: I would use the API modules instead of using axios directly here

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**